packages/babel-types/src/definitions/typescript.js

```diff
@@ -175,10 +176,11 @@ defineType("TSTypeReference", {
 
 defineType("TSTypePredicate", {
   aliases: ["TSType"],
-  visitor: ["parameterName", "typeAnnotation"],
+  visitor: ["parameterName", "typeAnnotation", "assertsModifier"],
   fields: {
+    assertsModifier: validateOptionalType("TSAssertsKeyword"),
```

Comment:
This should be a boolean, since the only needed information is if it is present or not.

Reply:
I was following the TypeScript AST.
But I understand Babel AST doesn't have to be the same as Typescript AST.